tests/storage_test.go

```diff
@@ -333,29 +362,71 @@ var _ = Describe("[Serial]Storage", func() {
-			table.DescribeTable("should be successfully started", func(newVMI VMICreationFunc, storageEngine string) {
-				tests.SkipPVCTestIfRunnigOnKindInfra()
-				var ignoreWarnings bool
+			Context("should be successfully", func() {
 				var pvName string
-				// Start the VirtualMachineInstance with the PVC attached
-				if storageEngine == "nfs" {
-					pvName = initNFS()
-					ignoreWarnings = true
-				} else {
-					pvName = tests.DiskAlpineHostPath
-				}
-				vmi = newVMI(pvName)
-				vmi = tests.RunVMIAndExpectLaunchWithIgnoreWarningArg(vmi, 120, ignoreWarnings)
+				var nfsPod *k8sv1.Pod
 
-				By("Checking that the VirtualMachineInstance console has expected output")
-				expecter, err := tests.LoggedInAlpineExpecter(vmi)
-				Expect(err).ToNot(HaveOccurred())
-				expecter.Close()
-			},
-				table.Entry("[test_id:3136]with Ephemeral PVC", tests.NewRandomVMIWithEphemeralPVC, ""),
-				table.Entry("[test_id:4619]with Ephemeral PVC from NFS", tests.NewRandomVMIWithEphemeralPVC, "nfs"),
-			)
+				AfterEach(func() {
+					if vmi != nil {
+						By("Deleting the VMI")
+						Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
+
+						By("Waiting for VMI to disappear")
+						tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
+
+						vmi = nil
+					}
+				})
+
+				AfterEach(func() {
+					if nfsPod != nil {
+						By("Deleting NFS pod")
+						Expect(virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Delete(tests.NFSTargetName, &metav1.DeleteOptions{})).To(Succeed())
+
+						By("Waiting for NFS pod to disappear")
+						tests.WaitForPodToDisappearWithTimeout(tests.NFSTargetName, 120)
+
+						nfsPod = nil
+					}
+				})
+
+				AfterEach(func() {
+					if pvName != "" && pvName != tests.DiskAlpineHostPath {
+						// PVs can't be reused
+						By("Deleting PV and PVC")
+						tests.DeletePvAndPvc(pvName)
+						pvName = ""
+					}
+				})
+
+				// The following case is mostly similar to the alpine PVC test above, except using different VirtualMachineInstance.
+				table.DescribeTable("started", func(newVMI VMICreationFunc, storageEngine string, family k8sv1.IPFamily) {
+					tests.SkipPVCTestIfRunnigOnKindInfra()
+					if family == k8sv1.IPv6Protocol {
+						libnet.SkipWhenNotDualStackCluster(virtClient)
+					}
+					var ignoreWarnings bool
+					// Start the VirtualMachineInstance with the PVC attached
+					if storageEngine == "nfs" {
+						nfsPod = initNFS()
+						pvName = createNFSPvAndPvc(family, nfsPod)
+						ignoreWarnings = true
+					} else {
+						pvName = tests.DiskAlpineHostPath
+					}
+					vmi = newVMI(pvName)
+					vmi = tests.RunVMIAndExpectLaunchWithIgnoreWarningArg(vmi, 120, ignoreWarnings)
+
+					By("Checking that the VirtualMachineInstance console has expected output")
+					expecter, err := tests.LoggedInAlpineExpecter(vmi)
+					Expect(err).ToNot(HaveOccurred())
+					expecter.Close()
```

Comment:
please consider using WaitUntilVMIReady which can get the expecter

Reply:
This code wasn't changed in the PR and it's not in the scope of the PR changing it.